python/ray/tune/tests/test_result_grid.py

```diff
@@ -224,6 +224,26 @@ def f(config):
     assert not any(key in representation for key in AUTO_RESULT_KEYS)
 
 
+def test_result_grid_repr(ray_start_2_cpus):
+    def f(config):
+        with tune.checkpoint_dir(step=0) as checkpoint_dir:
+            path = os.path.join(checkpoint_dir, "checkpoint")
+            with open(path, "w") as f:
+                f.write(json.dumps({"step": 0}))
+
```

Comment:
We can save a checkpiont with `session.report`. This `tune.checkpoint_dir` was the old API.
```suggestion
```